plugins/modules/s3_object.py

```diff
@@ -1011,7 +1011,8 @@ def main():
 
     if overwrite not in ['always', 'never', 'different', 'latest']:
         if module.boolean(overwrite):
-            overwrite = 'always'
+            # Fall to default configuration
+            overwrite = 'different'
```

Comment:
```suggestion
            overwrite = 'always'
```

`True` has been explicitly documented as equivalent to `always`, I think it's better to keep that for now.  I agree that the default when nothing is supplied should probably be `different` though.